src/mbgl/shader/pattern.vertex.glsl

```diff
@@ -9,6 +12,6 @@ varying vec2 v_pos_b;
 
 void main() {
     gl_Position = u_matrix * vec4(a_pos, 0, 1);
-    v_pos_a = (u_patternmatrix_a * vec3(a_pos, 1)).xy;
-    v_pos_b = (u_patternmatrix_b * vec3(a_pos, 1)).xy;
+    v_pos_a = (u_patternmatrix_a * vec3(a_pos, 1)).xy + u_offset_a;
+    v_pos_b = (u_patternmatrix_b * vec3(a_pos, 1)).xy + u_offset_b;
```

Comment:
I think we should remove u_patternmatrix_ like with did in -js. https://github.com/mapbox/mapbox-gl-native/issues/3488

Or we could re-add it in -js but I think the way -js does it now is slightly better.